src/include/duckdb/common/serializer/binary_deserializer.hpp

```diff
@@ -45,13 +45,17 @@ class BinaryDeserializer : public FormatDeserializer {
 	struct State {
 		uint32_t expected_field_count;
 		idx_t expected_size;
+		field_id_t expected_field_id;
 		uint32_t read_field_count;
-		State(uint32_t expected_field_count, idx_t expected_size)
-		    : expected_field_count(expected_field_count), expected_size(expected_size), read_field_count(0) {
+
+		State(uint32_t expected_field_count, idx_t expected_size, field_id_t expected_field_id)
+		    : expected_field_count(expected_field_count), expected_size(expected_size),
+		      expected_field_id(expected_field_id), read_field_count(0) {
```

Comment:
Also, it might be interesting to use `const` here to clarify which are mutable and which aren't... Sorry for all these `const` comments... 😅